common/k8stest/util_testpods.go

```diff
@@ -1218,3 +1218,70 @@ func GetMayastorPodNameonNodeByPrefix(prefix string, nodeName string) (string, e
 	}
 	return "", fmt.Errorf("failed to get mayastor pod with prefix %s on node %s", prefix, nodeName)
 }
+
+// GetCsiNodePodNameOnNodeByPrefix return csi node pod on a node with a given prefix
+func GetCsiNodePodNameOnNodeByPrefix(nodeName string) (string, error) {
+	podApi := gTestEnv.KubeInt.CoreV1().Pods
+	pods, err := podApi(common.NSMayastor()).List(context.TODO(), metaV1.ListOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to list pod in %s namespace, error: %v", common.NSMayastor(), err)
+	}
+	prefix := e2e_config.GetConfig().Product.CsiDaemonsetName
+	for _, pod := range pods.Items {
+		if strings.HasPrefix(pod.Name, prefix) {
+			if pod.Spec.NodeName == nodeName {
+				return pod.Name, nil
+			}
+		}
+	}
+	return "", fmt.Errorf("failed to get mayastor pod with prefix %s on node %s", prefix, nodeName)
+}
+
+func restartCsiNodePodsOnNode(nodeName string) error {
+	podApi := gTestEnv.KubeInt.CoreV1().Pods
+
+	podName, err := GetCsiNodePodNameOnNodeByPrefix(nodeName)
+
+	if err == nil {
+		logf.Log.Info("Restarting", "pod", podName)
+		time.Sleep(1 * time.Second)
+
+		delErr := podApi(common.NSMayastor()).Delete(context.TODO(), podName, metaV1.DeleteOptions{})
+		if delErr != nil {
+			logf.Log.Info("Failed to delete", "pod", podName, "error", delErr)
+			err = delErr
+		} else {
+			logf.Log.Info("Deleted", "pod", podName)
+		}
+	}
+	return err
+}
+
+// RestartMayastorPodsOnNode restart all mayastor pods scheduled on a given node
```

Comment:
Comment does not match the  function

Reply:
Comment should describe more of what the public function does, how the timeout parameters are used.

Reply:
[added comments](https://github.com/openebs/openebs-e2e/compare/97f6bb9f1285bb8f08fe736bb498948de52c0b3f..7a4e8d1886a4b1bebeba350d30dde9c35e2ec096)